<?xml version="1.0" encoding="utf-8"?>

<launch>
  
  <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
  
  <!-- Bag file path -->  
  <arg name="bag_file" value="$(find laser_mapper)/demonstration_files/kitti/2011_09_30_drive_18.bag"/>
  <rosparam param="LaserMapper/bag_file" subst_value="True">$(arg bag_file)</rosparam>
	
  <!-- Rviz visualization config file path -->
  <arg name="rviz_config_file" value="$(find laser_mapper)/demonstration_files/kitti/kitti_loop_closure.rviz"/>

  <!-- ROS parameters -->
  <rosparam>use_sim_time: true</rosparam>
  <rosparam command="load" file="$(find laser_mapper)/launch/kitti/kitti_loop_closure.yaml"/>

  <!-- rviz node -->
  <node name="visualizer" pkg="rviz" type="rviz" output="screen" respawn="$(arg respawn_value)" args="-d $(arg rviz_config_file)" />
  
  <!-- load classifier -->
  <arg name="classifier" value="$(find laser_mapper)/demonstration_files/kitti/random_forest_eigen_25trees.xml"/>
  <rosparam param="/LaserMapper/SegMatchWorker/SegMatch/Classifier/classifier_filename" subst_value="True">$(arg classifier)</rosparam>
  
  
  <!-- laser_mapper node -->
  <include file="$(find laser_mapper)/launch/laser_mapper.launch">
    <arg name="icp_configuration_file" value="$(find laser_mapper)/launch/kitti/icp_dynamic_outdoor.yaml"/>
    <arg name="icp_input_filters_file" value="$(find laser_mapper)/launch/kitti/input_filters_outdoor.yaml"/>
  </include>
  
    <!-- rosbag player node -->
  <node name="player" pkg="rosbag" type="play" output="screen" args=" -r 1 -s 0 --clock --pause $(arg bag_file)"/>	
 
</launch>
